### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by regislebrun**
This PR proposes several bug fixes and improvements:
  * The Normal and Student distribution had buggy parallel computeCDF() methods, calling the random generator in parallel. It has been fixed, and the getSample() and computeCDF() methods have been significantly improved. This work has been triggered by issue #1856 
  * The DiracCovarianceModel excluded the scale parameter from its parameters, resulting into a crash when used in the ProductCovarianceModel, see issue #1857 
  * The MaternModel was not always flagged as stationary depending on the constructor
  * The MarginalDistribution::computeCDF() method fails for some arguments due to a circular call in ContinuousDistribution::computeCDF()

### CODE REVIEW

**Comment by regislebrun** [comment]
I get a failure on a bonmin test, strange:
python: BonHeuristicDiveMIP.cpp:133: virtual int Bonmin::HeuristicDiveMIP::solution(double&, double*): Assertion `isNlpFeasible(minlp, primalTolerance)' failed.

Any idea on how to fix it?

**Comment by jschueller** [comment]
yes, its due to ipopt that got updated to 3.14.0, investigating

**Comment by jschueller** [comment]
turns out they changed the default of honor_original_bounds in ipopt, so we just have to change it back on our side, see #1849

**Review by sofianehaddad — APPROVED** [review]

File: lib/src/Uncertainty/Distribution/Normal.cxx [hunk 1]
```diff
@@ -379,6 +373,13 @@ Scalar Normal::computeCDF(const Point & point) const
   return value;
 } // computeCDF
 
+Sample Normal::computeCDF(const Sample & sample) const
+{
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/Normal.cxx:377]: you forgot to mention #1858

**Comment by jschueller** [comment]
There are commits from master here, it should probably be rebased.
I had already a commit about the missing KL include in #1766.

**Comment by regislebrun** [comment]
The remaining errors I get during the continuous integration are timeouts on other tests than mine...

**Comment by jschueller** [comment]
I rerun the CI, it should be ok.
I see the Changelog is untouched though.

**Pull request merged**